pkg/io/serialize.go

```diff
@@ -62,6 +62,10 @@ func Encode(writer io.Writer, values ...interface{}) (err error) {
 				return errors.WithMessage(err, "writing length of marshalled data")
 			}
 
+			// Nothing to be encoded when length is zero.
+			if length == 0 {
+				break
+			}
 			err = ByteSlice(data).Encode(writer)
```

Comment:
This is a rare case where we still have an `Encode` method. Would it make sense to replace that with a new method name?

Reply:
Are you referring to the `Encode` method on the `ByteSlice` type ?

Reply:
I am referring to `ByteSlice.Encode`. Is that terminology still suitable and consistent with the changes that we did?

Reply:
Yes, it is still consistent. Because, in case of binary types, we first Encode/Decode their length and then Encode/Decode them as ByteSlices.

We do so because, in case of binary types, the core abstraction which calls Encode/Decode does not know the length of data, as it could be different for different backends.

While, in case of ByteSlices, if the core abstraction is Encoding/Decoding them, it does know the length and hence the length need not be Encoded.